extensions/sql-database-projects/src/models/project.ts

```diff
@@ -635,34 +555,35 @@ export class Project implements ISqlProject {
 			return existingEntry;
 		}
 
-		// Update sqlproj XML
-		const fileEntry = this.createFileProjectEntry(normalizedRelativeFilePath, EntryType.File);
+		// Ensure the file exists // TODO: can be pushed down to DacFx
+		const absoluteFilePath = path.join(this.projectFolderPath, relativeFilePath);
+		await utils.ensureFileExists(absoluteFilePath, contents);
+
+		// Add the new script
+		let result: ResultStatus;
 
-		let xmlTag;
 		switch (itemType) {
 			case ItemType.preDeployScript:
-				xmlTag = constants.PreDeploy;
-				this._preDeployScripts.length === 0 ? this._preDeployScripts.push(fileEntry) : this._noneDeployScripts.push(fileEntry);
+				result = await this.sqlProjService.addPreDeploymentScript(this.projectFilePath, relativeFilePath);
+				await this.readPreDeployScripts();
+				await this.readNoneScripts();
 				break;
 			case ItemType.postDeployScript:
-				xmlTag = constants.PostDeploy;
-				this._postDeployScripts.length === 0 ? this._postDeployScripts.push(fileEntry) : this._noneDeployScripts.push(fileEntry);
+				result = await this.sqlProjService.addPostDeploymentScript(this.projectFilePath, relativeFilePath);
+				await this.readPostDeployScripts();
+				await this.readNoneScripts();
 				break;
 			default:
-				xmlTag = constants.Build;
-				this._files.push(fileEntry);
+				result = await this.sqlProjService.addSqlObjectScript(this.projectFilePath, relativeFilePath);
+				await this.readFilesInProject();
+				break;
 		}
 
-		const attributes = new Map<string, string>();
-
-		if (itemType === ItemType.externalStreamingJob) {
-			fileEntry.sqlObjectType = constants.ExternalStreamingJob;
-			attributes.set(constants.Type, constants.ExternalStreamingJob);
-		}
+		await this.readFolders();
```

Comment:
do files also need to be read?

Reply:
It's already being read a few lines above, in the `default` case, which is the only case that should affect the fileset since it's just sqlObjectScripts, right?

Reply:
ah ok oops I missed seeing that above